From 9448774028839a834a554384a7c64e89fd9c3486 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 4 Jan 2023 08:30:41 +0100
Subject: [PATCH] tests: fix tarball creation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Without this patch, the chmod 644 invocation takes away the executable
bit from the input_dir directory, thus preventing the chmod of the files
within the input_dir directory (due to lack of x-bit).

---
 testsuite/tar.tests | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testsuite/tar.tests b/testsuite/tar.tests
index 0f2e89112..18e571af3 100755
--- a/testsuite/tar.tests
+++ b/testsuite/tar.tests
@@ -67,7 +67,7 @@ testing "tar hardlinks and repeated files" '\
 ln input_hard1 input_hard2
 mkdir input_dir
 >input_dir/file
-chmod -R 644 *
+for f in *; do [ "$f" = "input_dir" ] || chmod -R 644 "$f"; done
 chmod    755 input_dir
 tar cf test.tar input input_dir/ input_hard1 input_hard2 input_hard1 input_dir/ input
 tar tvf test.tar | sed "s/.*[0-9] input/input/"
